src/charm.py

```diff
@@ -74,27 +90,26 @@ def on_start(self, event):
                         f"postgresql['enable'] = false",  # disable DB included in image
                         f"gitlab_rails['db_adapter'] = 'mysql2'",
                         f"gitlab_rails['db_encoding'] = 'utf8'",
-                        f"gitlab_rails['db_database'] = '{self.mysql.database}'",
-                        f"gitlab_rails['db_host'] = '{self.mysql.host}'",
-                        f"gitlab_rails['db_port'] = {self.mysql.port}",
-                        f"gitlab_rails['db_username'] = '{self.mysql.username}'",
-                        f"gitlab_rails['db_password'] = '{self.mysql.password}'",
+                        f"gitlab_rails['db_database'] = '{db.name}'",
+                        f"gitlab_rails['db_host'] = '{db.host}'",
+                        f"gitlab_rails['db_port'] = {db.port}",
+                        f"gitlab_rails['db_username'] = '{db.username}'",
+                        f"gitlab_rails['db_password'] = '{db.password}'",
                     ]),
                 }
             }],
         })
         self.state.is_started = True
-        unit.status = ActiveStatus()
+        self.on.serve_website.emit()
```

Comment:
Again, I'm not sure where an event is better than just a function call. It certainly is better if there is a state and you don't know who wants to observe it. (so interface/components would definitely want to use events, I'm not sure that the Charm objects would.)